UI/window-basic-main-outputs.cpp

```diff
@@ -761,90 +765,109 @@ bool SimpleOutput::SetupStreaming(obs_service_t *service)
 		auth->OnStreamConfig();
 
 	/* --------------------- */
-
-	const char *type = obs_service_get_output_type(service);
-	if (!type) {
-		type = "rtmp_output";
-		const char *url = obs_service_get_url(service);
-		if (url != NULL &&
-		    strncmp(url, FTL_PROTOCOL, strlen(FTL_PROTOCOL)) == 0) {
-			type = "ftl_output";
-		} else if (url != NULL && strncmp(url, RTMP_PROTOCOL,
-						  strlen(RTMP_PROTOCOL)) != 0) {
-			type = "ffmpeg_mpegts_muxer";
+	bool resetOutputs = true;
+	std::string streamOutputType;
+
+	for (auto &service : services) {
+		const char *type = obs_service_get_output_type(service);
+		if (!type) {
+			type = "rtmp_output";
+			const char *url = obs_service_get_url(service);
+			if (url != NULL && strncmp(url, FTL_PROTOCOL,
+						   strlen(FTL_PROTOCOL)) == 0) {
+				type = "ftl_output";
+			} else if (url != NULL &&
+				   strncmp(url, RTMP_PROTOCOL,
+					   strlen(RTMP_PROTOCOL)) != 0) {
+				type = "ffmpeg_mpegts_muxer";
+			}
 		}
-	}
 
-	/* XXX: this is messy and disgusting and should be refactored */
-	if (outputType != type) {
-		streamDelayStarting.Disconnect();
-		streamStopping.Disconnect();
-		startStreaming.Disconnect();
-		stopStreaming.Disconnect();
-
-		streamOutput = obs_output_create(type, "simple_stream", nullptr,
-						 nullptr);
-		if (!streamOutput) {
-			blog(LOG_WARNING,
-			     "Creation of stream output type '%s' "
-			     "failed!",
-			     type);
-			return false;
-		}
+		/* XXX: this is messy and disgusting and should be refactored */
+		if (outputType != type) {
+			if (resetOutputs) {
+				streamDelayStarting.Disconnect();
+				streamStopping.Disconnect();
+				startStreaming.Disconnect();
+				stopStreaming.Disconnect();
+				streamOutputs.clear();
+				resetOutputs = false;
+			}
```

Comment:
Outputs and their signals need to get cleared if the user switch to a service without redundancy.

Reply:
Moved this out of loop.